guides/keras_cv/random_walks_with_stable_diffusion.py

```diff
@@ -0,0 +1,382 @@
+## Interpolating Between Text Prompts
+
+In stable diffusion, a text prompt is encoded, and that encoding is used to
+guide the diffusion process. The latent manifold of this encoding has shape
+77x768 (that's huge!), and when we give StableDiffusion a text prompt, we're
+generating images from just one point in this manifold.
+
+To explore more of this manifold, we can interpolate between two text encodings
+and generate images at those interpolated points:
+"""
+
+prompt_1 = "A watercolor painting of a Golden Retriever at the beach"
+prompt_2 = "A still life DSLR photo of a bowl of fruit"
+interpolation_steps = 5
+
+encoding_1 = tf.squeeze(model.encode_text(prompt_1))
+encoding_2 = tf.squeeze(model.encode_text(prompt_2))
+
+interpolated_encodings = tf.linspace(encoding_1, encoding_2, interpolation_steps)
+
+# Show the size of the latent manifold
+print(encoding_1.shape)
+
+"""
+Once we've interpolated the encodings, we can generate images from each point.
+Note that in order to maintain some stability between the resulting images we
+keep the diffusion noise constant between images.
+"""
+
+seed = 12345
+noise = tf.random.normal((512 // 8, 512 // 8, 4), seed=seed)
+
+images = model.generate_image(
+    interpolated_encodings,
+    batch_size=interpolation_steps,
+    diffusion_noise=noise,
+)
+
+"""
+Now that we've generated some interpolated images, let's take a look at them!
+
+Throughout this tutorial, we're going to export sequences of images as gifs so
+that they can be easily viewed with some temporal context. For sequences of
+images where the first and last images don't match conceptually, we rubber-band
+the gif.
+
+If you're running in Colab, you can view your own GIFs by running:
+
+```
+from IPython.display import Image as IImage
+IImage("doggo-and-fruit-5.gif")
+```
+"""
+
+
+def export_as_gif(filename, images, frames_per_second=10, rubber_band=False):
+    if rubber_band:
+        images += images[2:-1][::-1]
+    images[0].save(
+        filename,
+        save_all=True,
+        append_images=images[1:],
+        duration=1000 // frames_per_second,
+        loop=0,
+    )
+
+
+export_as_gif(
+    "doggo-and-fruit-5.gif",
+    [Image.fromarray(img) for img in images],
+    frames_per_second=2,
+    rubber_band=True,
+)
+
+"""
+![Dog to Fruit 5](https://imgur.com/a/LHXceUi)
+
+The results may seem surprising. Generally, interpolating between prompts
+produces coherent looking images, and often demonstrate a progressive concept
+shift between the contents of the two prompts. This is indiciative of a high
+quality representation space.
+
+To best visualize this, we should do a much more fine-grained interpolation,
+using hundreds of steps. In order to keep batch size small (so that we don't
+OOM our accelerators), this requires manually batching our interpolated
+encodings.
+"""
+
+interpolation_steps = 160
+batch_size = 3
+
+encoding_step = (encoding_2 - encoding_1) / interpolation_steps
+encoding_step = tf.squeeze(encoding_step)
```

Comment:
I've realized that we can probably afford from a memory perspective to just generate the entire encoding batch here instead of doing the ugly intermediate interpolation steps.

Would be:

```
interpolated_encodings = tf.linspace(encoding1, encoding2, interpolation_steps)
encoding_batches = tf.split(interpolated_encodings, interpolation_steps // batch_size)

for index in range(interpolation_steps // batch_size):
    encodings = encoding_batches[index]
    images += [.....
```

I need to focus on benchmark tests rn but I can add this tomorrow afternoon if you don't get to it @LukeWood 


```

Reply:
Okay I've updated this and re-run all the experiments to verify it works.